models/POP/shell_scripts/cesm2_1/assimilate.csh

```diff
@@ -0,0 +1,358 @@
+# This block is an attempt to localize all the machine-specific
+# changes to this script such that the same script can be used
+# on multiple platforms. This will help us maintain the script.
+#=========================================================================
+
+echo "`date` -- BEGIN POP_ASSIMILATE"
+
+# As of CESM2.0, the assimilate.csh is called by CESM - and has
+# two arguments: the CASEROOT and the DATA_ASSIMILATION_CYCLE
+
+setenv CASEROOT $1
+setenv ASSIMILATION_CYCLE $2
+
+source ${CASEROOT}/DART_params.csh || exit 1
+
+# Python uses C indexing on loops; cycle = [0,....,$DATA_ASSIMILATION_CYCLES - 1]
+# "Fix" that here, so the rest of the script isn't confusing.
+@ cycle = $ASSIMILATION_CYCLE + 1
+
+#=========================================================================
+# Get the environmental variables needed for the script
+#=========================================================================
+cd ${CASEROOT}
+setenv CASE               `./xmlquery CASE             --value`
+setenv ENSEMBLE_SIZE      `./xmlquery NINST_OCN        --value`
+setenv EXEROOT            `./xmlquery EXEROOT          --value`
+setenv RUNDIR             `./xmlquery RUNDIR           --value`
+setenv ARCHIVE            `./xmlquery DOUT_S_ROOT      --value`
+setenv TOTALPES           `./xmlquery TOTALPES         --value`
+setenv STOP_N             `./xmlquery STOP_N           --value`
+
+#=========================================================================
+# Block 1: Determine time of model state ... from file name of first member
+# of the form "./${CASE}.pop_${ensemble_member}.r.2000-01-06-00000.nc"
+#
+# Piping stuff through 'bc' strips off any preceeding zeros.
+#=========================================================================
+cd ${RUNDIR}
+set FILE = `head -n 1 rpointer.ocn_0001.restart`
+set FILE = $FILE:r
+set OCN_DATE_EXT = `echo $FILE:e`
+set OCN_DATE     = `echo $FILE:e | sed -e "s#-# #g"`
+set OCN_YEAR     = `echo $OCN_DATE[1] | bc`
+set OCN_MONTH    = `echo $OCN_DATE[2] | bc`
+set OCN_DAY      = `echo $OCN_DATE[3] | bc`
+set OCN_SECONDS  = `echo $OCN_DATE[4] | bc`
+set OCN_HOUR     = `echo $OCN_DATE[4] / 3600 | bc`
+
+echo "valid time of model is $OCN_YEAR $OCN_MONTH $OCN_DAY $OCN_SECONDS (seconds)"
+echo "valid time of model is $OCN_YEAR $OCN_MONTH $OCN_DAY $OCN_HOUR (hours)"
+
+#=========================================================================
+# Block 2: Get observation sequence file ... or die right away.
+# The observation file names have a time that matches the stopping time of POP.
+#=========================================================================
+
+set YYYYMMDD = `printf %04d%02d%02d ${OCN_YEAR} ${OCN_MONTH} ${OCN_DAY}`
+set YYYYMM   = `printf %04d%02d     ${OCN_YEAR} ${OCN_MONTH}`
+set OBSFNAME = obs_seq.0Z.${YYYYMMDD}
+set OBS_FILE = ${BASEOBSDIR}/${YYYYMM}/${OBSFNAME}
+
+if (  -e   ${OBS_FILE} ) then
+   ${LINK} ${OBS_FILE} obs_seq.out
```

Comment:
I don't understand how this isn't failing on the second cycle.  The obs_seq.out already exists, and the LINK command does not (and should not) have the 'force' option specified.

Reply:
Yes, this code fails in the second cycle. I removed the line `${REMOVE} obs_seq.out` which was included because it copied the logic of `/models/cam-fv/shell_scripts/cesm2_1/assimilate.csh.template`